opengever/maintenance/scripts/fix_journal_message_mappings.py

```diff
@@ -0,0 +1,98 @@
+from ftw.journal.config import JOURNAL_ENTRIES_ANNOTATIONS_KEY
+from ftw.journal.interfaces import IAnnotationsJournalizable
+from ftw.upgrade import ProgressLogger
+from opengever.maintenance.debughelpers import setup_app
+from opengever.maintenance.debughelpers import setup_option_parser
+from opengever.maintenance.debughelpers import setup_plone
+from plone import api
+from zope.annotation.interfaces import IAnnotations
+from zope.i18nmessageid import Message
+import logging
+import sys
+import transaction
+
+
+logger = logging.getLogger('opengever.maintenance')
+handler = logging.StreamHandler(stream=sys.stdout)
+logging.root.addHandler(handler)
+logging.root.setLevel(logging.INFO)
+
+MAPPING_KEYS = ['title', 'repository']
+
+
+class JournalMessageFixer(object):
+    """Fixes encoding for zope.i18nmessageid.Message mapping values, for
+    journal entries of objects which has been created with an OGGBundle import.
+
+    See https://github.com/4teamwork/opengever.core/pull/3369 for more details.
+    """
+
+    def __init__(self, portal, options):
+        self.portal = portal
+        self.options = options
+        self.counter = 0
+
+    def get_journal_entries(self, obj):
+        """Returns all journal entries for the given object.
+        """
+        if IAnnotationsJournalizable.providedBy(obj):
+            annotations = IAnnotations(obj)
+            return annotations.get(JOURNAL_ENTRIES_ANNOTATIONS_KEY, [])
+
+        return []
+
+    def fix_entries(self, obj):
+        for entry in self.get_journal_entries(obj):
+            message = entry['action'].get('title')
```

Comment:
do we always have an `action`? if no you may wanna do 

```python
message = entry.get('action', {}).get('title')
```
instead.

Reply:
Yes, we always have an `action. If not the script should fail, so that I can have a closer look to those objects.